plugin/phpactor.vim

```diff
@@ -242,6 +303,54 @@ function! phpactor#_selectionEnd()
     return line2byte(lineEnd) + columnEnd -1
 endfunction
 
+function! phpactor#_applyTextEdits(path, edits)
+    call phpactor#_switchToBufferOrEdit(a:path)
+
+    let postCursorPosition = getpos('.')
+
+    for edit in a:edits
+
+        let newLines = 0
+
+        " start = { line: 1234, character: 0 }
+        let start = edit['start']
+
+        " end = { line: 1234, character: 0 }
+        let end = edit['end']
+
+        " move the cursor into the start position
+        call setpos('.', [ 0, start['line'] + 1, start['character'] + 1 ])
+
+        if start['character'] != 0 || end['character'] != 0
+            throw "Non-zero character offsets not supported in text edits, got " . json_encode(edit)
+        endif
+
+        " to delete
+        let linesToDelete = end['line'] - start['line']
+        if linesToDelete > 0
+            call execute('normal ' . linesToDelete . 'dd')
+        endif
+
+        if edit['text'] == "\n"
+            " if this is just a new line, add a new line
+            call append(start['line'], '')
+            let newLines = 1
+        else
+            " insert characters after the current line
+            let appendLines = split(edit['text'], "\n")
+            call append(start['line'], appendLines)
+            let newLines = newLines + len(appendLines)
+        endif
+
+        if postCursorPosition[1] > start['line']
+            let postCursorPosition[1] = postCursorPosition[1] + newLines
+        endif
```

Comment:
This seems to work fine - but couldn't figure out how to do this "properly" in VIM script, doesn't matter for the naieve implementation currently supported.